src/Halogen/HTML/Events/Forms.purs

```diff
@@ -1,48 +1,34 @@
--- | Convenience functions for working with form elements.
-
+-- | Convenience functions for working with form events.
 module Halogen.HTML.Events.Forms
-  ( onValueChanged
+  ( onValueChange
+  , onValueInput
   , onChecked
-  , onInput
   ) where
 
 import Prelude
 
-import DOM
-
-import Data.Maybe
-import Data.Either
-import Data.Foreign
-import Data.Foreign.Class
-import Data.Traversable (traverse)
+import Data.Either (either)
+import Data.Foreign (toForeign)
+import Data.Foreign.Class (IsForeign, readProp)
 
-import Control.Plus
-import Control.Alternative
+import Halogen.HTML.Core (Prop(), eventName, handler)
+import Halogen.HTML.Events.Handler (EventHandler())
 
-import Halogen.HTML.Events.Handler
+-- | Attach event handler to event `key` with getting `prop` field as an
+-- | argument of `handler`.
+addForeignPropHandler :: forall f value. (Applicative f, IsForeign value) => String -> String -> (value -> EventHandler (f Unit)) -> Prop (f Unit)
```

Comment:
That's what I'm thinking, it would be nice if there were a `purescript-events` library that has an on-opinionated, low-level abstraction across all browsers, which unifies the common stuff and in some type safe fashion supports the useful browser-specific stuff. Halogen and other libs could depend on it.